tree/ntuple/v7/inc/ROOT/RNTupleView.hxx

```diff
@@ -149,17 +149,40 @@ private:
    FieldT fField;
    /// Used as a Read() destination for fields that are not mappable
    RFieldBase::RValue fValue;
+   /// Whether the user supplied an external memory location to read the value into
+   bool fUserProvidedAddress{false};
 
-   RNTupleView(DescriptorId_t fieldId, Internal::RPageSource *pageSource)
-      : fField(pageSource->GetSharedDescriptorGuard()->GetFieldDescriptor(fieldId).GetFieldName()),
-        fValue(fField.CreateValue())
+   void SetupField(DescriptorId_t fieldId, Internal::RPageSource *pageSource)
    {
       fField.SetOnDiskId(fieldId);
       Internal::CallConnectPageSourceOnField(fField, *pageSource);
       if ((fField.GetTraits() & RFieldBase::kTraitMappable) && fField.HasReadCallbacks())
          throw RException(R__FAIL("view disallowed on field with mappable type and read callback"));
    }
 
+   RNTupleView(DescriptorId_t fieldId, Internal::RPageSource *pageSource)
+      : fField(pageSource->GetSharedDescriptorGuard()->GetFieldDescriptor(fieldId).GetFieldName()),
+        fValue(fField.CreateValue())
+   {
+      SetupField(fieldId, pageSource);
+   }
+
+   RNTupleView(DescriptorId_t fieldId, Internal::RPageSource *pageSource, std::shared_ptr<T> objPtr)
+      : fField(pageSource->GetSharedDescriptorGuard()->GetFieldDescriptor(fieldId).GetFieldName()),
+        fValue(fField.BindValue(objPtr))
+   {
+      fUserProvidedAddress = true;
+      SetupField(fieldId, pageSource);
+   }
+
+   RNTupleView(DescriptorId_t fieldId, Internal::RPageSource *pageSource, T *rawPtr)
+      : fField(pageSource->GetSharedDescriptorGuard()->GetFieldDescriptor(fieldId).GetFieldName()),
+        fValue(fField.BindValue(std::shared_ptr<T>(std::shared_ptr<void>(), rawPtr)))
```

Comment:
I'm not sure this works, because we alias the shared pointer to a temporary shared pointer. I think we need a static dummy shared pointer instead.

Reply:
I took a second read at https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr and the aliasing constructor I am using here is actually a C++20 feature 
```
For the second overload taking an rvalue, r is empty and r.get() == nullptr after the call.(since C++20)
```

Reply:
I will use the static shared_ptr approach